### PR TITLE
chore: tile ify dashboard tiles

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

We want to add text tiles to dashboards. Dashboards currently assume all tiles are insights. 

## Changes

Operate on tiles as much as possible to decouple dashboards and insights

Currently the DB relationship is

```
[dashboard] <-one to many-> [dashboard tile] <-- many to one --> [insight]
```

but the API collapses tile properties on to Insights and the front end deals with

```
[dashboard] <---> [insights]
```

In preparation for adding different types of cards to dashboards we need to loosen that relationship

This is a step along the road to text cards on dashboards, spiked here https://github.com/PostHog/posthog/pull/11924 and https://github.com/PostHog/posthog/pull/11377

## How did you test this code?

adding developer tests and running the site locally

Things to test

- [x] move insight to dashboard

current state: removed undo. can follow up after adding text cards

move-to-dashboard

- [x] duplicate insight from insight view

duplicate-insight-correctly

- [x] duplicate insight from card list view

duplicate from card view

- [x] duplicate insight from list view

duplicate from list view

- [x] duplicate insight from dashboard

duplicate from dashbaord

- [x] remove insight from dashboard

remove from dashboard

- [x] add insight to dashboard

add to dashboard

- [x] delete dashboard

delete dashboard

- [x] duplicate dashboard

NB existing "bug" where when we duplicate a dashboard we duplicate its insights

duplicate dashboard

- [x] set card color

setting-colors-without-moving

- [x] set card layout

set-layout

- [x] update insight updates dashboard

update-insight-date-range

- [x] rename insight

rename-working

### CODE REVIEW

**Review by Twixes — COMMENTED** [review]
Some small comments but this seems sensible overall

File: posthog/api/dashboard.py [hunk 1]
```diff
@@ -271,6 +311,21 @@ def retrieve(self, request: Request, *args: Any, **kwargs: Any) -> response.Resp
         serializer = DashboardSerializer(dashboard, context={"view": self, "request": request})
         return response.Response(serializer.data)
 
+    @action(methods=["PATCH"], detail=True)
+    def move_tile(self, request: Request, *args: Any, **kwargs: Any) -> response.Response:
```
[the remark this answers is not in the document]
Comment (pauldambra) [reply posthog/api/dashboard.py:315]: It's used in dashboardLogic `moveToDashboard`

[the remark this answers is not in the document]
Comment (Twixes) [reply posthog/api/dashboard.py:315]: Ahh, I missed that because GitHub hid `dashboardLogic.tsx` due to diff size 😅

**Review by Twixes — COMMENTED** [review]
.

**Review by Twixes — APPROVED** [review]
Ich lichten

**Pull request merged**